### PR TITLE
DOC: Update roadmap for completions

### PR DESCRIPTION
**Pull request by TomAugspurger**
Moves "Improved documentation" to completed section.

I don't think any others can be marked as done yet, though "Numba-accelerated operations" may be getting close (is that right @mroeschke?).

### CODE REVIEW

**Comment by mroeschke** [comment]
Regarding Numba, pretty much complete in reference to the `and in groupby and window contexts` line in that section.
Will need to see if it's still relevant for the non groupby and window contexts

**Comment by jbrockmendel** [comment]
ci/../doc/source/development/roadmap.rst:193:Heading capitalization formatted incorrectly. Please correctly capitalize "Completed Items" to "Completed items"

File: doc/source/development/roadmap.rst [hunk 1]
```diff
@@ -203,3 +189,20 @@ should be notified of the proposal.
 When there's agreement that an implementation
 would be welcome, the roadmap should be updated to include the summary and a
 link to the discussion issue.
+
+Completed Items
```
Comment (TomAugspurger) [inline doc/source/development/roadmap.rst:193]:
```suggestion
Completed items
```

**Comment by TomAugspurger** [comment]
Fixed.

**Comment by jreback** [comment]
great

**Pull request merged**